sound/soc/sof/intel/hda.h

```diff
@@ -913,4 +913,53 @@ extern struct sdw_intel_ops sdw_callback;
 struct sof_ipc4_fw_library;
 int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
 			      struct sof_ipc4_fw_library *fw_lib, bool reload);
+
+/**
+ * struct hda_dai_widget_dma_ops - DAI widget DMA ops optional by default unless specified otherwise
+ * @stream_get_dma: Function pointer to get DAI DMA data
+ * @stream_put_dma: Function pointer to put DAI DMA data
+ * @setup: Function pointer for DAI DMA setup
+ * @free: Function pointer for DAI DMA free
+ * @pre_trigger: Function pointer for DAI DMA pre-trigger actions
+ * @trigger: Function pointer for DAI DMA trigger actions
+ * @post_trigger: Function pointer for DAI DMA post-trigger actions
+ */
+struct hda_dai_widget_dma_ops {
+	void *(*stream_get_dma)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+				struct snd_pcm_substream *substream);
+	void (*stream_put_dma)(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+			       struct snd_pcm_substream *substream);
```

Comment:
We call this what it is now? get_hext_stream and return `struct hdac_ext_stream *` ?